pwnlib/tubes/process.py

```diff
@@ -288,7 +291,10 @@ def __init__(self, argv = None,
         self.executable = executable_val
 
         #: Environment passed on envp
-        self.env = os.environ if env is None else env_val
+        self.env = dict(os.environ) if env is None else env_val
```

Comment:
I would follow the principle of least surprise here, not mutating the arguments:
```suggestion
        self.env = dict(os.environ if env is None else env_val)
```